### PR TITLE
Added documentation and removed methods we added but no longer use

### PR DESCRIPTION
**Pull request by alex-stanescu**
Adds some documentation so that we get better feedback from our PR

### CODE REVIEW

File: src/include/catalog/catalog_accessor.h [hunk 1]
```diff
@@ -371,19 +380,10 @@ class CatalogAccessor {
...
-  * @return TransactionContext to be used for creating and building an index
+  * @return The transaction that this accessor was initialized with
   */
   common::ManagedPointer<transaction::TransactionContext> GetTransactionContext() const;
 
-  /**
-   * Instantiates a new accessor into for the same catalog with the new txn
-   * @param new_txn the new transaction
-   */
-  CatalogAccessor SetTxn(const common::ManagedPointer<transaction::TransactionContext> new_txn) const {
```
Comment (kjobanputra) [inline src/include/catalog/catalog_accessor.h:383]: Why did you take this out?

File: src/include/catalog/catalog_accessor.h [hunk 2]
```diff
@@ -400,10 +400,6 @@ class CatalogAccessor {
         default_namespace_(postgres::NAMESPACE_DEFAULT_NAMESPACE_OID) {}
 
 
-  bool TransferLock(common::ManagedPointer<transaction::TransactionContext> from, common::ManagedPointer<transaction::TransactionContext> to) {
```
Comment (kjobanputra) [inline src/include/catalog/catalog_accessor.h:403]: Also, why did you take this out?

**Review by kjobanputra — APPROVED** [review]

File: src/include/catalog/catalog_accessor.h (hunk 1 above)
Comment (alex-stanescu) [reply src/include/catalog/catalog_accessor.h:383]: we dont use it

File: src/include/catalog/catalog_accessor.h (hunk 2 above)
Comment (alex-stanescu) [reply src/include/catalog/catalog_accessor.h:403]: we dont use it

**Pull request merged**